pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -1201,6 +1206,22 @@ func TestPendingPodsMetric(t *testing.T) {
 		}
 		pInfos = append(pInfos, p)
 	}
+	totalWithDelay := 20
+	var pInfosWithDelay = make([]*framework.PodInfo, 0, totalWithDelay)
```

Comment:
instead, you could make the creation of podInfos a function that accepts the current timestamp.

Then, you would have the same steps that you are setting (`add, step, add`), and some operands like:
`pInfos(0, 20), nil, pInfos(20, 30)`

Reply:
but the pInfos will be calculated before the execution of steps ` (add, step, add),`, so it can't get the current timestamp dynamically

Reply:
I've added a new func to avoid duplicate code

Reply:
well operands could be functions instead of actual values, but this is good enough, I guess.